zerver/lib/actions.py

```diff
@@ -5244,6 +5244,7 @@ def check_invite_limit(realm: Realm, num_invitees: int) -> None:
 def do_invite_users(user_profile: UserProfile,
                     invitee_emails: SizedTextIterable,
                     streams: Iterable[Stream],
+                    invite_expires_in: int=settings.INVITATION_LINK_VALIDITY_DAYS,
```

Comment:
Oh, I see, hadn't realized this field was new in your PR.  Couldn't we just have this be accessed `prereg_user.confirmation.` rather than having two different expiration fields?

Reply:
Yeah, we can store expiration duration in `PreregistrationUser` and use that field to check expiration of `confirmation key`, this way we can avoid two different fields.